src/Safebox.sol

```diff
@@ -0,0 +1,238 @@
+    address public recipient;
+    /// @notice Reference to the new recipient when it is to be changed.
+    address public pendingRecipient;
+
+    /**
+     * @notice `usr` was granted owner access.
+     * @param usr The user address.
+     */
+    event Rely(address indexed usr);
+    /**
+     * @notice `usr` owner access was revoked.
+     * @param usr The user address.
+     */
+    event Deny(address indexed usr);
+    /**
+     * @notice `usr` was granted custodian access.
+     * @param usr The user address.
+     */
+    event Hope(address indexed usr);
+    /**
+     * @notice The owner withdrawn assets from the safebox.
+     * @param token The token withdrawn.
+     * @param amount The amount withdrawn.
+     */
+    event Withdraw(address indexed token, uint256 amount);
+    /**
+     * @notice A deposit was made into the safebox.
+     * @param token The token deposited.
+     * @param amount The amount deposited.
+     */
+    event Deposit(address indexed token, uint256 amount);
+    /**
+     * @notice A contract parameter was updated.
+     * @param what The changed parameter name. The supported values are: "recipient".
+     * @param data The new value of the parameter.
+     */
+    event File(bytes32 indexed what, address data);
+    /**
+     * @notice The recipient has been sucessfully changed.
+     * @param recipient The new recipient address.
+     */
+    event RecipientChange(address indexed recipient);
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "Safebox/not-authorized");
+        _;
+    }
+
+    /**
+     * @param _vat The MCD vat module.
+     * @param _owner The safebox owner.
+     * @param _custodian The safebox custodian.
+     * @param _recipient The recipient for assets in the safebox.
+     */
+    constructor(address _vat, address _owner, address _custodian, address _recipient) {
+        require(_recipient != address(0), "Safebox/invalid-recipient");
+
+        wards[_owner] = 1;
+        emit Rely(_owner);
+
+        can[_custodian] = 1;
+        emit Hope(_custodian);
+
+        vat = VatLike(_vat);
+        recipient = _recipient;
+    }
+
+    /*//////////////////////////////////
+            MakerDAO Interfaces
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` owner access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` owner access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"recipient"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "recipient") {
```

Comment:
This should be blocked when VAT is not live.

Reply:
++

Reply:
Done

Reply:
Fixed.